File: 0880-decoded-string-at-index/0880-decoded-string-at-index.java
class Solution {
    public String decodeAtIndex(String s, int k) {
        long len = 0;
        for(int i=0; i<s.length(); i++) {
            if(s.charAt(i) >= '0' && s.charAt(i) <= '9') {
                len *= (s.charAt(i) - '0');
            }
            else {
                len++;
            }
        }
        for(int i=s.length()-1; i>=0; i--) {
            k = (int) (k%len);
            if(k == 0 && !Character.isDigit(s.charAt(i))) {
                return Character.toString(s.charAt(i));
            }
            else if(s.charAt(i) >= '0' && s.charAt(i) <= '9') len /= (s.charAt(i) - '0');
            else len--;
        }
        return "";
    }
}

File: 0880-decoded-string-at-index/0880-decoded-string-at-index.cpp
class Solution {
public:
    string decodeAtIndex(string s, int k) {
        long long len = 0;
        for(char c:s) {
            if(c >= '2' && c <= '9') len *= (c - '0');
            else len++;
        }
        for(int i=s.size()-1; i>=0; i--) {
            k = k%len;
            if(k == 0 && !isdigit(s[i])) {
                string ans = "";
                ans += s[i];
                return ans;
            }
            else if(isdigit(s[i])) {
                len /= (s[i] - '0');
            }
            else {
                len--;
            }
        }
        return "";
    }
};